src/main/java/teammates/ui/servlets/RequestTraceFilter.java

```diff
@@ -78,11 +80,16 @@ public void doFilter(ServletRequest req, ServletResponse resp, FilterChain chain
         RequestTracer.init(traceId, spanId, timeoutInSeconds);
 
         Map<String, Object> requestDetails = new HashMap<>();
+        Map<String, Serializable> requestParams = HttpRequestHelper.getRequestParameters(request);
         requestDetails.put("requestMethod", request.getMethod());
         requestDetails.put("requestUrl", request.getRequestURI());
         requestDetails.put("userAgent", request.getHeader("User-Agent"));
-        requestDetails.put("requestParams", HttpRequestHelper.getRequestParameters(request));
+        requestDetails.put("requestParams", requestParams);
         requestDetails.put("requestHeaders", HttpRequestHelper.getRequestHeaders(request));
+        Map<String, Object> userInfo = new HashMap<>();
+        userInfo.put("regkey", requestParams.get(Const.ParamsNames.REGKEY));
+        userInfo.put("email", requestParams.get(Const.ParamsNames.STUDENT_EMAIL));
```

Comment:
This is a wrong usage. `STUDENT_EMAIL` does not always correspond to the user who calls the action (in fact, half of its usages are not).

On a general note, a system that assumes user information from "raw" query parameter like this is asking for trouble. Regkey is not "raw" in that sense because it is random sequence of alphanumeric strings, but this is as raw and human-readable as it can be.

Reply:
Understood. The email field is removed. So for now only regkey is logged if it is present.